bundles/http/Citrix/5c20233e-ead7-4385-a3f9-c703f194514b/5b0d4f89-2cfb-499b-8f63-f70cbed8fa13.js

```diff
@@ -0,0 +1,136 @@
+            "incrementalSyncFunction": updateComments
+        }
+    ],
+    "model": {
+        "tables": [
+            {
+                "name": "updated_comments",
+                "columns": [
+                    {
+                        "name": "id",
+                        "type": "INTEGER",
+                        "primaryKey": true
+                    },
+                    {
+                        "name": "body",
+                        "type": "STRING",
+                        "length": 100
+                    },
+                    {
+                        "name": "created_at",
+                        "type": "DATETIME"
+                    },
+                    {
+                        "name": "commenter_id",
+                        "type": "INTEGER"
+                    },
+                    {
+                        "name": "user_name",
+                        "type": "STRING"
+
+                    },
+                    {
+                        "name": "user_email",
+                        "type": "STRING"
+                    }
+                ]
+            }
+        ]
+    }
+});
+
+function replaceBodyCharacters(key, value) {
+    let str = value;
+    if (key === "body") {
+        str = str.replace(/<[^>]+>/gm, '');
+        str = str.replace('\n&nbsp;', '');
+    }
+    return str;
+}
+
+async function updateComments({dataStore, client, latestSynchronizationTime}) {
+    let incidentComments = "";
+    let queryFields = "";
+    let exit = false;
+    let latestSync = new Date(latestSynchronizationTime);
+
+    if (latestSynchronizationTime === undefined)
+        queryFields = '?per_page=' + LIMIT + '&page=';
+    else {
+        queryFields = '?q=lastUpdateDate>=' + moment(latestSync).format('YYYY-MM-DD') + '&per_page=' + LIMIT + '&page=';
+    }
+
+    //Get page count
+    let urlEndpoint = ENDPOINT + queryFields + 0;
+    let resp = await client.fetch(urlEndpoint);
+    const pages = resp.ok && resp.headers["map"]["x-total-pages"] ? resp.headers["map"]["x-total-pages"] : 1000;
+
+    //Loop through all pages
+    for (j = 0; j < pages; j++) {
+
+        if (exit) {
+            break;
+        }
+
+        let incidentsEndpoint = ENDPOINT + queryFields + j;
+
+        let resp = await client.fetch(incidentsEndpoint);
+
+        if (resp.ok) {
+            var incidents = await resp.json();
+
+            if (incidents.length === 0) {
+                exit = true;
+            }
+
+            //Loop through each incident
+            for (i = 0; i < incidents.length; i++) {
+
+                let commentEndpoint = "";
+
+                if (latestSynchronizationTime === undefined)
+                    commentEndpoint = ENDPOINT + '/' + incidents[i].id + '/comments';
+                else {
+                    commentEndpoint = ENDPOINT + '/' + incidents[i].id + '/comments' + '?q=updated_at>=' + moment(latestSync).format('YYYY-MM-DD');
+                }

```

Comment:
```suggestion
                let commentEndpoint = `/incidents/${incidents[i].id}/comments`;

                if (!latestSynchronizationTime)
                    commentEndpoint = `${commentEndpoint}?q=updated_at>=${moment(latestSync).format('YYYY-MM-DD')}`;
                }
```

Reply:
You can prepare the whole thing before `for` loops
```js
const updatedAt = latestSynchronizationTime ? '' : `?q=updated_at>=${moment(latestSynchronizationTime).format('YYYY-MM-DD')}`;
```

and here you can do just
```js
var response = await client.fetch(`/incidents/${incidents[i].id}/comments${updatedAt}`);
```

Reply:
Done.

Reply:
I see.
The condition which I suggested here has the opposite meaning here and you use it in the latest change.
Sorry for confusing you.